### ISSUE TITLE
Make working directory used by ProcessRunner configurable

### ISSUE THREAD

**Issue by siegfriedpammer** [daveaglick/Buildalyzer#233]
Some things like detection of the current .NET version from global.json depend on the working directory of the msbuild call. Currently it is not possible to explicitly set the working directory of the msbuild invocation inside the Buildalyzer build.

Proposed API:

```cs
   public class EnvironmentOptions
    {
...
        public string WorkingDirectory { get; set; }
...
    }
```

(and also pass it to `class BuildEnvironment` of course)

and change

https://github.com/daveaglick/Buildalyzer/blob/5b732a8ce572efbba077dd08871fb609c69f94ce/src/Buildalyzer/ProjectAnalyzer.cs#L167

to

```cs
 using (ProcessRunner processRunner = new ProcessRunner(fileName, arguments, buildEnvironment.WorkingDirectory ?? Path.GetDirectoryName(ProjectFile.Path), GetEffectiveEnvironmentVariables(buildEnvironment), Manager.LoggerFactory)) 
```

What do you think?

**Comment by Corniel** [comment]
As a user of this lib, I did never needed it (yet), but only seems fair that is configurable.

**Comment by daveaglick** [comment]
Good suggestion, especially since a lot of MS Build functionality depends on having a correct working directory and different scenarios may change that.

**Comment by daveaglick** [comment]
This is now implemented and will go out with the 6.0.1 release. Thanks for the suggestion!
